app/Jobs/Ninja/TaskScheduler.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+namespace App\Jobs\Ninja;
+
+
+use App\Jobs\Report\SendToAdmin;
+use App\Libraries\MultiDB;
+use App\Models\Company;
+use App\Models\ScheduledJob;
+use App\Models\Scheduler;
+use Carbon\Carbon;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+
+class TaskScheduler implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+
+    public function __construct()
+    {
+
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+
+        $pending_schedulers = $this->fetchJobs();
+        foreach ($pending_schedulers as $scheduler) {
+            $this->doJob($scheduler);
+        }
+    }
+
+    private function doJob(Scheduler $scheduler)
+    {
+        $job = $scheduler->job;
+
+        $company = Company::find($job->company_id);
+        if (!$job || !$company) {
+            return;
+        }
+        MultiDB::setDb($company->db);
```

Comment:
This needs to be the first line of the method, otherwise we may be hitting the wrong database for the $company entity query.